hack/compare-k8s-feature-gates.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+#
+# SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+usage() {
+  echo "Usage:"
+  echo "> compare-k8s-feature-gates.sh [ -h | <old version> <new version> ]"
+  echo
+  echo ">> For example: compare-k8s-feature-gates.sh 1.26 1.27"
```

Comment:
The example doesn't work. For which K8s versions exactly this script works? Could you please properly document it? For me the script only works for the versions `1.32 1.33`.

Reply:
What about also introducing a check for the minimum supported version, similar to `hack/compute-k8s-controllers.sh`, eg: 
https://github.com/gardener/gardener/blob/a1898ec38fb075cdf1c6622a9039686ed7963c08/hack/compute-k8s-controllers.sh#L83-L87

Reply:
I like the suggestion!

Reply:
@ialidzhikov suggested 1.32+. Should we adapt the suggestion like this?
```bash
 for version in "${versions[@]}"; do 
   if [ "$version" \< "1.32" ]; then 
     echo "Versions less than 1.32 are not supported." 
     exit 1 
   fi 
```

Reply:
Great! Let's keep the note in the usage docs and enforce it with the above suggested validation.
Thanks!